jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/RegexPathSpec.java

```diff
@@ -38,34 +59,80 @@ public RegexPathSpec(String regex)
             declaration = regex;
         int specLength = declaration.length();
         // build up a simple signature we can use to identify the grouping
-        boolean inGrouping = false;
+        boolean inTextList = false;
+        boolean inQuantifier = false;
         StringBuilder signature = new StringBuilder();
 
         int pathDepth = 0;
+        char last = 0;
         for (int i = 0; i < declaration.length(); i++)
         {
             char c = declaration.charAt(i);
             switch (c)
             {
-                case '[':
-                    inGrouping = true;
+                case '^': // ignore anchors
+                case '$': // ignore anchors
+                case '\'': // ignore escaping
+                case '(': // ignore grouping
+                case ')': // ignore grouping
                     break;
-                case ']':
-                    inGrouping = false;
+                case '+': // single char quantifier
+                case '?': // single char quantifier
+                case '*': // single char quantifier
+                case '|': // alternate match token
+                case '.': // any char token
                     signature.append('g'); // glob
                     break;
-                case '*':
+                case '{':
+                    inQuantifier = true;
+                    break;
+                case '}':
+                    inQuantifier = false;
+                    break;
+                case '[':
+                    inTextList = true;
+                    break;
+                case ']':
+                    inTextList = false;
                     signature.append('g'); // glob
                     break;
                 case '/':
-                    if (!inGrouping)
+                    if (!inTextList && !inQuantifier)
                         pathDepth++;
                     break;
```

Comment:
Should be there a `case` for `<` and `>` (for named groups)?

Reply:
Not needed, the group detects a signature `g` properly and that sets up the signature check later approriately.
Other uses of `<` and `>` are allowed.
New test case added.